### PR TITLE
CA-267553: Fix offline XC showing up to date bug

### PR DESCRIPTION
**Pull request by jijiang**
The bug is that when XC cannot connect to Internet, By Server Update
page shows all the hosts up to date.

Signed-off-by: Ji Jiang <ji.jiang@citrix.com>

### CODE REVIEW

**Comment by GaborApatiNagy** [comment]
It looks good to me, but I'm wondering whether we should store the IsFull flags in each row? It is always the same for all rows in the grid.
By the way, isFull is not quite meaningful to me, this name could be misleading in the future.

File: XenAdmin/TabPages/ManageUpdatesPage.cs [hunk 1]
```diff
@@ -383,16 +407,26 @@ private void UpdateDetails()
...
+                            if (IsFullRow)
+                            {
+                                _patchingStatusCell.Value = outOfDate
+                                    ? Properties.Resources._000_error_h32bit_16
+                                    : Properties.Resources._000_Tick_h32bit_16;
+                                _statusCell.Value = outOfDate ? Messages.NOT_UPDATED : Messages.UPDATED;
+                            }
+                            else
+                            {
+                                _statusCell.Value = "";
```
Comment (GaborApatiNagy) [inline XenAdmin/TabPages/ManageUpdatesPage.cs]: It's nicer to have string.Empty instead of ""

File: XenAdmin/TabPages/ManageUpdatesPage.cs [hunk 2]
```diff
@@ -1361,9 +1401,11 @@ private string GetHostUpdateDetailsCsvQuotes(IXenConnection xenConnection, Host
             Program.Invoke(Program.MainWindow, delegate
             {
                 pool = hasPool ? Helpers.GetPool(xenConnection).Name() : String.Empty;
-                requiredUpdates = RequiredUpdatesForHost(host);
+                requiredUpdates = CheckForUpdateSucceeded ? RequiredUpdatesForHost(host) : String.Empty;
```
Comment (MihaelaStoica) [inline XenAdmin/TabPages/ManageUpdatesPage.cs]: I am not sure that the invoking on the UI thread is needed on line 1401

**Comment by MihaelaStoica** [comment]
This looks good now. Merging.

**Pull request merged**